typescript/sdk/src/deploy/verify/ContractVerifier.ts

```diff
@@ -81,13 +91,43 @@ export class ContractVerifier extends MultiGeneric<VerificationInput> {
     if (isGetRequest) {
       response = await fetch(`${apiUrl}?${params}`);
     } else {
-      response = await fetch(apiUrl, {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/x-www-form-urlencoded',
-        },
-        body: params,
-      });
+      // For Blockscout explorers, we need to ensure module and action are always query params
+      if (family === ExplorerFamily.Blockscout) {
+        const formData = new URLSearchParams();
+        const urlWithParams = new URL(apiUrl);
+        urlWithParams.searchParams.append('module', 'contract');
+        urlWithParams.searchParams.append('action', action);
+
+        // remove any extraneous keys from body
+        for (const [key, value] of params) {
+          switch (key) {
+            case 'module':
+            case 'action':
+            case 'licenseType':
+            case 'apikey':
+              break;
+            default:
+              formData.append(key, value);
+              break;
+          }
```

Comment:
I think originally I had some more complex logic happening depending on the block explorer type, but that wasn't necessary in the end - however the switch remained

nonetheless you'll be happy to know this will change with the refactor PR https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3255/files#diff-085d8c3137fce2e849ef557083b77087824cb0f33b5801cbd97b4f7cdf718f52